python/ray/experimental/serve/global_state.py

```diff
@@ -100,27 +101,50 @@ def init_or_get_http_server(self,
         if "http_server" not in self.actor_handle_cache:
             [handle] = ray.get(
                 self.actor_nursery_handle.start_actor.remote(
-                    HTTPActor, init_args=(), tag="http_server"))
+                    HTTPActor, init_args=(), init_kwargs={},
+                    tag="http_server"))
             handle.run.remote(host=host, port=port)
             self.refresh_actor_handle_cache()
         return self.actor_handle_cache["http_server"]
 
-    def init_or_get_router(self):
-        if "queue_actor" not in self.actor_handle_cache:
+    def _get_queueing_policy(self, default_policy):
+        return_policy = default_policy
+        # check if there is already a queue_actor running
+        # with policy as p.name for the case where
+        # serve nursery exists: ray.experimental.get_actor(SERVE_NURSERY_NAME)
+        for p in Policy:
+            queue_actor_tag = "queue_actor::" + p.name
+            if queue_actor_tag in self.actor_handle_cache:
+                return_policy = p
+                break
+        return return_policy
+
+    def init_or_get_router(self,
+                           queueing_policy=Policy.Random,
+                           policy_kwargs={}):
+        # get queueing policy
+        self.queueing_policy = self._get_queueing_policy(
+            default_policy=queueing_policy)
+        queue_actor_tag = "queue_actor::" + self.queueing_policy.name
+        if queue_actor_tag not in self.actor_handle_cache:
             [handle] = ray.get(
                 self.actor_nursery_handle.start_actor.remote(
-                    CentralizedQueuesActor, init_args=(), tag="queue_actor"))
+                    self.queueing_policy.value,
```

Comment:
if queuing_policy is user supplied class, we don't need to extract `.value` here.

Reply:
This is still needed to check if there is already there nursery of ray actors. 
`# Try to get serve nursery if there exists
    try:
        ray.experimental.get_actor(SERVE_NURSERY_NAME)
        global_state = GlobalState()
        return
    except ValueError:
        pass`

Reply:
`.value` is needed to know `actor_cls`

Reply:
yes we need `.value` for the enum case, can you add support such that the self.queue_policy can be a class as well?